### PR TITLE
[query] import_google_sheet

### PR DESCRIPTION
**Pull request by Aleisha02**

### CODE REVIEW

File: hail/python/hail/methods/impex.py [hunk 1]
```diff
@@ -2979,3 +2979,66 @@ def import_avro(paths, *, key=None, intervals=None):
...
+        'Home State': str
+        'Major': str
+        'Extracurricular Activity': str
+    ----------------------------------------
+    Key: []
+    ----------------------------------------
+
+    Parameters
+    ----------
+    spreadsheetID: :class:`str` or :obj:`list` of :obj:`str`
```
Comment (danking) [inline hail/python/hail/methods/impex.py:3013]: The spreadsheet id can only be a str, not a list of str, so let's change this to: 
```
spreadsheetID: :class:`str`
```

File: hail/python/hail/methods/impex.py [hunk 2]
```diff
@@ -2979,3 +2979,66 @@ def import_avro(paths, *, key=None, intervals=None):
...
+        File containing sheet to import.
```
Comment (danking) [inline hail/python/hail/methods/impex.py:3014]: Let's describe this as the spread sheet id. Can you include an example above of what a Google Sheet URL and the corresponding ID looks like? I think most people will have trouble finding the ID on the first try.

File: hail/python/hail/methods/impex.py [hunk 3]
```diff
@@ -2979,3 +2979,66 @@ def import_avro(paths, *, key=None, intervals=None):
...
+            **{column_name: ht['cells'][index]}
+        )
+        index = index + 1
+    ht = ht.add_index('index')
+    ht = ht.filter(ht['index'] != 0)
+    ht = ht.drop('cells')
+    ht = ht.drop('index')
+
+    return ht
+
```
Comment (danking) [inline hail/python/hail/methods/impex.py:3044]: Can you delete this empty line at the end of the file?

File: hail/src/main/scala/is/hail/expr/ir/StringTableReader.scala [hunk 4]
```diff
@@ -157,3 +157,4 @@ class StringTableReader(
                                 PField("text", PCanonicalString(true), 1)), true).subsetTo(requestedType.rowType).asInstanceOf[PStruct],
      PCanonicalStruct.empty(required = true))
 }
+
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/StringTableReader.scala:160]: Can you delete this empty line?

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 5]
```diff
@@ -2999,3 +2983,21 @@ case class RelationalLetTable(name: String, value: IR, body: TableIR) extends Ta
...
+    (jv \ "name").extract[String] match {
+      case "TableNativeReader" => TableNativeReader.fromJValue(fs, jv)
+      case "TableFromBlockMatrixNativeReader" => TableFromBlockMatrixNativeReader.fromJValue(fs, jv)
+      case "StringTableReader" => StringTableReader.fromJValue(fs, jv)
+      case "AvroTableReader" => AvroTableReader.fromJValue(jv)
+      case "GoogleSheetReader" => GoogleSheetReader.fromJValue(fs, jv)
+      case _ => jv.extract[TableReader]
+    }
+  }
+}
```
Comment (danking) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala:3003]: Can you move this back to around line 102?

**Review by danking — DISMISSED** [review]
This looks awesome. I just have a comment about the docs and a few nitpicks.

**Comment by Aleisha02** [comment]
Ready for another look

**Review by danking — APPROVED** [review]
LGTM!

**Comment by danking** [comment]
grr I think this betrothed.

**Comment by danking** [comment]
bitrotted*

**Comment by danking** [comment]
Waiting on google to explain why the latest version of the sheets library gives me 403s.

**Comment by danking** [comment]
No explanation from Google, but I figured out how to make it work. TLDR: don't use the thing that seems like the obviously correct choice, use some obscure other approach.

**Comment by danking** [comment]
@tpoterba I heavily modified this so best to get a second review.

**Review by danking — APPROVED** [review]

**Pull request closed without merging**